pkg/cache/config.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package cache
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+type cacheType string
+
+const (
+	timeDay              = 24 * time.Hour
+	defaultCacheValidity = 5 * timeDay
+	cacheTypeEmpty       = cacheType("")
+	cacheTypeNone        = cacheType("none")
+	cacheTypeSQLite      = cacheType("sqlite")
+)
+
+// Config holds the cache configuration.
+type Config struct {
+	// Type is the type of the cache.
+	Type cacheType
+	// Validity is the duration for which the cache is valid.
+	Validity time.Duration
+	// Jitter is the jitter to apply when considering a cached entry valid or not.
+	Jitter time.Duration
+
+	cacheParser *configParser
+}
+
+// NewConfig is the constructor for Config.
+func NewConfig() Config {
+	return Config{
+		cacheParser: newConfigParser(),
+	}
+}
+
+// Present tell whether a cache configuration is present.
+func (c *Config) Present() bool {
+	return c.Type != cacheTypeNone
+}
+
+// UnmarshalYAML puts the unmarshalled yaml data into the internal cache parser
+// struct. This prevents access to the string data of jitter and validity.
+func (c *Config) UnmarshalYAML(value *yaml.Node) error {
+	if err := value.Decode(c.cacheParser); err != nil {
+		return err
+	}
+	if err := c.load(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// load validates the cache configuration from the parser and copy it
+// into the configuration.
+func (c *Config) load() error {
+	switch c.cacheParser.Type {
```

Comment:
Can we lower case the Type here, so we can make this change but also without breaking change?

Reply:
@bwplotka I can un-export the `Type`, but I need a getter to see the actual type. What's the intent of this suggested change?

Reply:
Ah, nevermind. I was confused.